### ISSUE TITLE
Memory Leak Regex v5.32.1 

### ISSUE THREAD

**Issue by perlscallion** [Perl/perl5#18602]
There's a memory leak here, Perl v5.32.1 (as part of Fedora 33)

#!/usr/bin/perl

   $string = "STRING 1";

   while (1)
      {
      $x = test_leak($string, "string 1");
      $x = test_leak($string, "string 2");
      }
   exit;

   sub test_leak
   {
local ($string, $match) = @_;

   $match =~ s/\s/\\s/g;  # This Leaks
#  $match =~ s/\s/ /g;    # This doesn't Leak

   return ($string =~ /$match/i);
   }
~

**Comment by perlscallion** [comment]
No... That cut and paste didn't work. Maybe too much magic pasting backslashes. Trying again...

$match =~ s/\s/\\s/g; # This Leaks

**Comment by perlscallion** [comment]
No. Definitely too much magic. Should be read as

$match =~ s/<Backslash>s/<Backslash><Backslash>s/g;

**Comment by perlscallion** [comment]
Mad, that didn't work either....

**Comment by perlscallion** [comment]
Can try this...

``#!/usr/bin/perl

   $string = "STRING 1";

   while (1)
      {
      $x = test_leak($string, "string 1");
      $x = test_leak($string, "string 2");
      }
   exit;

   sub test_leak
   {
local ($string, $match) = @_;

   $match =~ s/\s/\\s/g;  # This Leaks
#  $match =~ s/\s/ /g;    # this doesn't Leak

   return ($string =~ /$match/i);
   }
~

**Comment by perlscallion** [comment]
No chance....

**Comment by perlscallion** [comment]
Maybe this time...

[perl-leak.txt](https://github.com/Perl/perl5/files/6030433/perl-leak.txt)

**Comment by perlscallion** [comment]
Got there. Leaving this issue closed, I'll try again from the beginning

**Comment by karenetheridge** [comment]
this might help -- https://guides.github.com/features/mastering-markdown/

tldr: use three backticks on a separate line to begin and end code blocks.
